src/login/components/SwamidErrors/Errors.js

```diff
@@ -0,0 +1,112 @@
+import React, { useEffect, useState } from "react";
+import { useLocation } from "react-router-dom";
+import "../../styles/index.scss";
+import i18n from "../../translation/InjectIntl_HOC_factory";
+import { swamidErrorData } from "./swamidErrorData";
+
+function Errors(props){
+  let query = new URLSearchParams(useLocation().search);
+  const [techInformation, setTechInformation] = useState({});
+  useEffect(()=> {
+    let errorurl_ts = query.get("errorurl_ts");
+    let errorurl_rp = query.get("errorurl_rp");
+    let errorurl_tid = query.get("errorurl_tid");
+    let errorurl_ctx = query.get("errorurl_ctx");
+    //Convert unix time stamp to (YYYY-MM-DD) format
+    const convertUnixDate = new Date(errorurl_ts * 1000);
+    const newDateUrlTs = 
+      ' ('+ convertUnixDate.getFullYear()+'-' + ((convertUnixDate.getMonth()+1).toString())+ '-'+(convertUnixDate.getDate().toString())+')';
+
+      setTechInformation({
+        errorurl_ts: errorurl_ts &&  errorurl_ts !== "ERRORURL_TS" ? errorurl_ts + newDateUrlTs : undefined, 
+        errorurl_rp: errorurl_rp && errorurl_rp !== "ERRORURL_RP" ? errorurl_rp : undefined,
+        errorurl_tid: errorurl_tid && errorurl_tid !== "ERRORURL_TID" ? errorurl_tid : undefined,
+        errorurl_ctx: errorurl_ctx && errorurl_ctx !== "ERRORURL_CTX" ? errorurl_ctx : undefined,
+    })
+  }, []);
+
+  let isSpecificError = "";
+  let errorurlCode = query.get("errorurl_code");
+  let errorurlCtx = query.get("errorurl_ctx");
+  let errorurlRp = query.get("errorurl_rp");
+  let common = swamidErrorData.common;
+  let specialRp = swamidErrorData["sp.ladok.se"];
+
+  const checkErrorUrlCtx = () => {
+    //Compare error url ctx query string and swaimidErrorData.common 
+    Object.keys(common).map(key=>{
+      if(key === errorurlCode){
+      let result = common[key];
+      Object.keys(result).map((urlCtx)=> {
+        if(errorurlCtx && errorurlCtx.includes(urlCtx)){
+          return isSpecificError = (
+            <div className="specific-error">{props.translate(Object.values(result[urlCtx]).toString())}</div>
+        )}
+      })
+    }
+  })};
+
+  if(errorurlRp && errorurlRp.includes("sp.ladok.se"))
```

Comment:
We can't use hardcoded special SPs anywhere.

Can we combine the two lookups to something like this perhaps?

``` js
let messages = swamidErrorData.common;
if swamidErrorData.has(errorurlRp) {
    messages = swamidErrorData[errorurlRp]
}

const checkErrorUrlCtx = (messages) => {
    Object.keys(messages).map(key=>{
        if(key === errorurlCode){
            let result = messages[key];
            Object.keys(result).map((urlCtx)=> {
                if(errorurlCtx && errorurlCtx.includes(urlCtx)){
                    return isSpecificError = (
                        <div className="specific-error">{props.translate(Object.values(result[urlCtx]).toString())}</div>
                    )} else {
                    return checkErrorUrlCtx(swamidErrorData.common)
                }
            })
        } else {
            return checkErrorUrlCtx(swamidErrorData.common)
        }
    })};
```

Reply:
I tried combining the two lookups but it causes an error "RangeError: Maximum call stack size exceeded"
so I kept the code separated